### PR TITLE
Adding preliminary SSZ encoding/decoding of Transaction Network Payload

### PR DESCRIPTION
**Pull request by gezero**
Signed-off-by: Jiri Peinlich <jiri.peinlich@gmail.com>

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/ssz/SignedBlobTransaction.java [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+    SSZUtil.Uint64SSZWrapper gas = new SSZUtil.Uint64SSZWrapper();
+    MaybeAddress address = new MaybeAddress();
+    SSZUtil.Uint256SSZWrapper value = new SSZUtil.Uint256SSZWrapper();
+    Data data = new Data(MAX_CALL_DATA_SIZE);
+
+    SSZUtil.SSZVariableSizeList<AccessTuple> accessList =
+        new SSZUtil.SSZVariableSizeList<>(AccessTuple::new);
+    SSZUtil.Uint256SSZWrapper maxFeePerData = new SSZUtil.Uint256SSZWrapper();
+
+    SSZUtil.SSZFixedSizeList<VersionedHash> blobVersionedHashes =
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/ssz/SignedBlobTransaction.java:88]: is checked somewhere that list size is <= `MAX_VERSIONED_HASHES_LIST_SIZE`?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/ssz/TransactionNetworkPayload.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import org.apache.tuweni.ssz.SSZReader;
+import org.apache.tuweni.ssz.SSZWriter;
+
+public class TransactionNetworkPayload implements SSZUtil.SSZType {
+  public static final int FIELD_ELEMENTS_PER_BLOB = 4096;
+  SignedBlobTransaction transaction = new SignedBlobTransaction();
+  SSZUtil.SSZFixedSizeList<KZGCommitment> kzgCommitments =
+      new SSZUtil.SSZFixedSizeList<>(KZGCommitment::new);
+  SSZUtil.SSZFixedSizeList<Blob> blobSSZVariableSizeList =
+      new SSZUtil.SSZFixedSizeList<>(Blob::new);
```
Comment (fab-10) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/ssz/TransactionNetworkPayload.java:30]: should checks for max list size be added?

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java [hunk 3]
```diff
@@ -72,7 +132,7 @@ public static Transaction decodeForWire(
       final Bytes typedTransactionBytes = rlpInput.readBytes();
       final TransactionType transactionType =
           TransactionType.of(typedTransactionBytes.get(0) & 0xff);
-      return getDecoder(transactionType).decode(RLP.input(typedTransactionBytes.slice(1)));
+      return getDecoder(transactionType).decode(typedTransactionBytes.slice(1));
```
[the remark this answers is not in the document]
Comment (gezero) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java:135]: yes, i'll add that

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java [hunk 4]
```diff
@@ -38,24 +40,82 @@
...
+  public static Transaction decodeBlob(final SSZReader input, final int length) {
+
+    TransactionNetworkPayload payload = new TransactionNetworkPayload();
+    payload.decodeFrom(input, length);
+
+    SignedBlobTransaction signedBlobTransaction = payload.getTransaction();
+
+    final SignedBlobTransaction.BlobTransaction blobTransaction =
+        signedBlobTransaction.getMessage();
+    return Transaction.builder()
```
[the remark this answers is not in the document]
Comment (gezero) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/encoding/TransactionDecoder.java]: should be there now

**Pull request closed without merging**